src/radixsort.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+
+from __future__ import unicode_literals, division
+from collections import deque
+from math import log10
+
+
+def _get_digit(num, position):
+    """Find digit at particular decimal position in a number."""
+    return num // 10 ** position % 10
+
+
+def _put_nums_in_buckets(a_list, position):
+    """
+    Queue numbers into appropriate buckets based on the value
+    at the particular decimal position.
+    """
+    q_list = []
+    for x in range(10):
+        queue_ = deque()
+        q_list.append(queue_)
+    for num in a_list:
+        digit = _get_digit(num, position)
+        q_list[digit].append(num)
+    return q_list
+
+
+def _find_num_of_iterations(a_list):
+    """
+    Find number of the sorting iterations to be performed
+    (this equals number of digits in the biggest number in the list).
+    """
+    num_of_iterations = int(log10(max(a_list)) + 1)
+    return num_of_iterations
+
+
+def _dequeue_queues_into_list(q_list):
+    """Dequeue all the buckets in the pending list into a list."""
+    a_list = []
```

Comment:
You could also use `itertools.chain` to replace this entire function -- it is an awesome tool which will flatten an iterable of iterables into a single iterable. Check it out in the Python docs.